### PR TITLE
St/beaconing extensibinility

### PR DESCRIPTION
**Pull request by SeyedaliTaba**
This PR explains the research question of how to optimise routes based on various and ever-changing criteria.

### CODE REVIEW

File: draft-meynell-panrg-scion-research-questions.md [hunk 1]
```diff
@@ -228,6 +259,29 @@ Examples for requesting CORE segments between different ISDs or within an ISD (a
 | 64-0:0:0    | 64-2:0:13   | 60                |
 {: #segment-count-example title="CORE segment count examples"}
 
+## Periodic beacon propagation
+* Is it necessary to propagate beacons periodically? If so, why?
```
Comment (knmeynell) [inline draft-meynell-panrg-scion-research-questions.md:263]:
```suggestion
* Is it necessary to propagate beacons periodically? 
```

File: draft-meynell-panrg-scion-research-questions.md [hunk 2]
```diff
@@ -228,6 +259,29 @@ Examples for requesting CORE segments between different ISDs or within an ISD (a
 | 64-0:0:0    | 64-2:0:13   | 60                |
 {: #segment-count-example title="CORE segment count examples"}
 
+## Periodic beacon propagation
+* Is it necessary to propagate beacons periodically? If so, why? 
+  * For path freshness, only the origin needs to originate beacons periodically, and others disseminate immediately.
+  * For response to link failures or availability of new paths, beacon services can respond instantly.
+* Re-propgating the same set of beacons at each interval is a waste of resources The periodic propagation can be used for the discovery of new paths at each interval, enhancing the scalability, and path diversity.
```
Comment (knmeynell) [inline draft-meynell-panrg-scion-research-questions.md:266]:
```suggestion
* Re-propagating the same set of beacons at each interval makes inefficient use of resources. The periodic propagation can be used for the discovery of new paths at each interval, enhancing the scalability and path diversity.
```

File: draft-meynell-panrg-scion-research-questions.md [hunk 3]
```diff
@@ -228,6 +259,29 @@ Examples for requesting CORE segments between different ISDs or within an ISD (a
...
+
+## Beacon optimization and beaconing extensibility
```
Comment (knmeynell) [inline draft-meynell-panrg-scion-research-questions.md:268]:
```suggestion
## Beacon optimization and extensibility
```

File: draft-meynell-panrg-scion-research-questions.md [hunk 4]
```diff
@@ -228,6 +259,29 @@ Examples for requesting CORE segments between different ISDs or within an ISD (a
...
+* Communication quality requirements vary based on source, destination, and applications.
```
Comment (knmeynell) [inline draft-meynell-panrg-scion-research-questions.md:269]:
```suggestion
* Communication requirements vary according to source, destination, and application. Satisfying all these requirements either requires discovering all paths in the network, or optimizing paths during the beaconing process. Selecting the 5 shortest paths per destination cannot satisfy all requirements.
```

File: draft-meynell-panrg-scion-research-questions.md [hunk 5]
```diff
@@ -228,6 +259,29 @@ Examples for requesting CORE segments between different ISDs or within an ISD (a
...
+* Satisfying all these criteria either requires discovering all paths in the network, or optimizing paths during the beaconing process.
```
Comment (knmeynell) [inline draft-meynell-panrg-scion-research-questions.md:270]:
```suggestion
```

File: draft-meynell-panrg-scion-research-questions.md [hunk 6]
```diff
@@ -228,6 +259,29 @@ Examples for requesting CORE segments between different ISDs or within an ISD (a
...
+  * Clearly, the 5-shortest path per destination cannot satisfy all communication requirements.
```
Comment (knmeynell) [inline draft-meynell-panrg-scion-research-questions.md:271]:
```suggestion
```

File: draft-meynell-panrg-scion-research-questions.md [hunk 7]
```diff
@@ -228,6 +259,29 @@ Examples for requesting CORE segments between different ISDs or within an ISD (a
...
+* What optimization functions should be applied to beacons when propagating them?
```
Comment (knmeynell) [inline draft-meynell-panrg-scion-research-questions.md:272]:
```suggestion
* What optimization functions should be applied to beacons and what metrics should be considered when propagating them ?
```

File: draft-meynell-panrg-scion-research-questions.md [hunk 8]
```diff
@@ -228,6 +259,29 @@ Examples for requesting CORE segments between different ISDs or within an ISD (a
...
+  * What are the optimality metrics considered in these functions?
```
Comment (knmeynell) [inline draft-meynell-panrg-scion-research-questions.md:273]:
```suggestion
```

File: draft-meynell-panrg-scion-research-questions.md [hunk 9]
```diff
@@ -228,6 +259,29 @@ Examples for requesting CORE segments between different ISDs or within an ISD (a
...
+* Who should select these functions?
+* How should the deployment of these functions be enforced?
+* How can multiple functions be applied concurrently for different src/dst/applications?
```
Comment (knmeynell) [inline draft-meynell-panrg-scion-research-questions.md:276]:
```suggestion
* How can multiple functions be applied concurrently for different source and destination applications?
```

File: draft-meynell-panrg-scion-research-questions.md [hunk 10]
```diff
@@ -228,6 +259,29 @@ Examples for requesting CORE segments between different ISDs or within an ISD (a
...
+* How should end ASes express their desired requirements to the inter-domain control plane?
+* How do these requirements translate into concrete optimization functions?
+* How does standardization of the functions work? 
+* Criteria change over time.
```
Comment (knmeynell) [inline draft-meynell-panrg-scion-research-questions.md:280]:
```suggestion
* Criteria changing over time.
```

**Review by knmeynell — APPROVED** [review]

**Comment by nicorusti** [comment]
@SeyedaliTaba the tool does not like trailing whitespaces, you can use `make fix-lint` ti make the draft compile

**Review by nicorusti — COMMENTED** [review]
Hi Ali, thanks a lot for this initial text. I've added some specific comments above, then some more general comments: 
- I feel that overall the key question about beaconing is to improve the tradeoffs between scalability/finding sufficient paths and doing it quickly. I'd try to highlight that in the text
- This topics connects to question 2.7  of RFC9217 (and also a bit 2.8)
  > Competing control inputs from path-aware endpoints and the routing control plane may lead to more difficult traffic engineering or non-convergent forwarding, especially if the endpoints' and operators' notion of the "best" path for given traffic diverges significantly. 
  
  I'd mention that, and that this has implications for traffic engineering. This is a question that we get a lot at IETF. I'd almost call this paragraph `Beacon optimization and traffic engineering` if this could fit..

File: draft-meynell-panrg-scion-research-questions.md [hunk 11]
```diff
@@ -228,6 +259,28 @@ Examples for requesting CORE segments between different ISDs or within an ISD (a
 | 64-0:0:0    | 64-2:0:13   | 60                |
 {: #segment-count-example title="CORE segment count examples"}
 
+## Periodic beacon propagation
+* Is it necessary to propagate beacons periodically? 
+  * For path freshness, only the origin needs to originate beacons periodically, and others disseminate immediately.
+  * For response to link failures or availability of new paths, beacon services can respond instantly.
+* Re-propagating the same set of beacons at each interval makes inefficient use of resources. The periodic propagation can be used for the discovery of new paths at each interval, enhancing the scalability and path diversity.
```
[the remark this answers is not in the document]
Comment (juagargi) [reply draft-meynell-panrg-scion-research-questions.md]: 1. A newly connected AS first sets its topology offline. Its neighbors have to do the same.
2. The new AS learns about the other border router presence through BFD.
3. The neighbor's border router will consider this interface as active.
4. The neighbor's control plane will use this interface to propagate beacons.
5. The new AS receives new PCBs and can propagate and build paths.

**Review by nicorusti — APPROVED** [review]

**Pull request merged**